rllib/policy/sample_batch.py

```diff
@@ -906,26 +906,30 @@ def _slice(self, slice_: slice) -> "SampleBatch":
             # Build our slice-map, if not done already.
             if not self._slice_map:
                 sum_ = 0
-                for i, l in enumerate(self[SampleBatch.SEQ_LENS]):
-                    for _ in range(l):
-                        self._slice_map.append((i, sum_))
-                    sum_ += l
```

Comment:
Python's built-in type `int` is immutable, it's okay to use `x += y`. Statement `x += y` will create a new `int` instance.

```python
>>> x = 1000

>>> x
1000
>>> id(x)
140102686402576

>>> x += 1
>>> x
1001
>>> id(x)
140102686403152
```

Here `l` and `sum_` are tensors, `sum_ += l` will not create a new instance. `slice_map.append((i, sum_))` is adding the reference to the same instance `sum_` in memory.

```python
>>> import torch
>>> sum_ = 0
>>> l = torch.ones((), dtype=torch.int32)

>>> sum_
0
>>> id(sum_)
94230645960128

>>> sum_ += l
>>> sum_
tensor(1, dtype=torch.int32)
>>> id(sum_)
140100675095104

>>> sum_ += l
>>> sum_
tensor(2, dtype=torch.int32)
>>> id(sum_)
140100675095104
```